Move `@bazel_tools//tools/jdk:JacocoCoverage` dependency to runtime dep list

In rules_kotlin, we are listing `@bazel_tools//tools/jdk:JacocoCoverage` as a [compile time dep](https://github.com/bazelbuild/rules_kotlin/blob/cbc4ced96a6685236c398cc2554fa4abefec1087/src/main/kotlin/io/bazel/kotlin/builder/tasks/BUILD.bazel#L31) and that will pull jacoco into kotlin release package which results in the version of jacoco (and other bazel tools) being dependent on `rules_kotlin`'s bazel version. 

For example, if a project using latest Bazel 6.0.0 (jacoco version is 0.8.7) but the kotlin rule is built by bazel 4.2 (jacoco version is 0.8.3). Then when we are running `bazel coverage` on this project, java code will use jacoco 0.8.7 and kotlin code will use jacoco 0.8.3, which is incorrect.

To fix this, we should move the jacoco dep to runtime dependency list if possible.

This causes coverage to fail when running under Bazel 5 with this error:

```
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Constructor.call(CallerImpl.kt:41)
	at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:108)
	at io.kotest.engine.spec.InstantiateSpecKt.javaReflectNewInstance(instantiateSpec.kt:42)
	... 34 more
Caused by: java.lang.NoClassDefFoundError: org/jacoco/agent/rt/internal_1f1cc91/Offline
	at com.whiteswandata.betfairapi.AsyncBetfairWrapperKtTest.$jacocoInit(AsyncBetfairWrapperKtTest.kt)
	at com.whiteswandata.betfairapi.AsyncBetfairWrapperKtTest.<init>(AsyncBetfairWrapperKtTest.kt)
	... 41 more
Caused by: java.lang.ClassNotFoundException: org.jacoco.agent.rt.internal_1f1cc91.Offline
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 43 more
```

This is because `rules_kotlin` was built with Bazel 4.2.0, bundles Bazel 4.2.0's version of Jacoco and uses it to instrument the compiled Kotlin code, but Bazel 5 includes a new version and uses that at runtime, and the mismatch leads to this crash.

A workaround is to rebuild `rules_kotlin` with Bazel 5 so it includes the correct version of Jacoco, but it would be better if `rules_kotlin` used the Jacoco version from the Bazel version it is running with, not the one it was built with.

Note that only the new version of Jacoco in Bazel 5 includes fixes for Kotlin 1.5 (https://github.com/jacoco/jacoco/issues/1155), so currently the only way to get coverage for Kotlin 1.5 is to use Bazel 5 and a custom build of `rules_kotlin` built with Bazel 5 (I have tested this and it works).

--- a/src/main/kotlin/BUILD.release.bazel
+++ b/src/main/kotlin/BUILD.release.bazel
@@ -45,7 +45,10 @@ java_binary(
     ],
     main_class = "io.bazel.kotlin.builder.cmd.Build",
     visibility = ["//visibility:public"],
-    runtime_deps = [":worker"],
+    runtime_deps = [
+        ":worker",
+        "@bazel_tools//tools/jdk:JacocoCoverage",
+    ],
 )
 
 java_binary(

--- a/src/main/kotlin/io/bazel/kotlin/builder/tasks/BUILD.bazel
+++ b/src/main/kotlin/io/bazel/kotlin/builder/tasks/BUILD.bazel
@@ -19,6 +19,9 @@ kt_bootstrap_library(
     srcs = glob([
         "**/*.kt",
     ]),
+    neverlink_deps = [
+        "@bazel_tools//tools/jdk:JacocoCoverage",
+    ],
     visibility = ["//src:__subpackages__"],
     deps = [
         "//src/main/kotlin/io/bazel/kotlin/builder/toolchain",
@@ -28,7 +31,6 @@ kt_bootstrap_library(
         "//src/main/protobuf:deps_java_proto",
         "//src/main/protobuf:kotlin_model_java_proto",
         "//src/main/protobuf:worker_protocol_java_proto",
-        "@bazel_tools//tools/jdk:JacocoCoverage",
         "@com_github_jetbrains_kotlin//:kotlin-preloader",
         "@kotlin_rules_maven//:com_google_protobuf_protobuf_java",
         "@kotlin_rules_maven//:com_google_protobuf_protobuf_java_util",

--- a/src/test/kotlin/io/bazel/kotlin/builder/BUILD
+++ b/src/test/kotlin/io/bazel/kotlin/builder/BUILD
@@ -47,6 +47,7 @@ java_library(
         "@kotlin_rules_maven//:com_google_protobuf_protobuf_java",
     ],
     runtime_deps = [
+        "@bazel_tools//tools/jdk:JacocoCoverage",
         "@com_github_jetbrains_kotlin//:kotlin-reflect",
         "@com_github_jetbrains_kotlin//:kotlin-stdlib",
     ],

